src/stack/methods.jl

```diff
@@ -47,8 +47,10 @@ Apply function `f` to each layer of the `stacks`.
 If `f` returns `DimArray`s the result will be another `DimStack`.
 Other values will be returned in a `NamedTuple`.
 """
-Base.map(f, s::AbstractDimStack...) = _maybestack(s[1], map(f, map(NamedTuple, s)...))
-Base.map(f, s::Union{AbstractDimStack,NamedTuple}...) = _maybestack(_firststack(s...), map(f, map(NamedTuple, s)...))
+function Base.map(f, s1::Union{AbstractDimStack,NamedTuple}, s::Union{AbstractDimStack,NamedTuple}...)
+    results = map(f, map(NamedTuple, (s1, s...)...))
```

Comment:
Seems to cause a stack overflow. Perhaps this will break the infinite recursion by always peeling off one argument?
```suggestion
    results = map(f, NamedTuple(s1), map(NamedTuple, s)...))
```